src/host/getset.cpp

```diff
@@ -359,9 +361,27 @@ void ApiRoutines::GetNumberOfConsoleMouseButtonsImpl(ULONG& buttons) noexcept
 
         const auto oldMouseMode{ WI_IsFlagSet(context.InputMode, ENABLE_MOUSE_INPUT) };
         const auto newMouseMode{ WI_IsFlagSet(mode, ENABLE_MOUSE_INPUT) };
-        if (oldMouseMode != newMouseMode)
+        const auto newQuickEditMode{ WI_IsFlagSet(gci.Flags, ENABLE_QUICK_EDIT_MODE) };
+
+        // Whether we ask the terminal for mouse input is decided by 2 flags, ENABLE_MOUSE_INPUT
+        // and ENABLE_QUICK_EDIT_MODE
+        // If ENABLE_MOUSE_INPUT is set and ENABLE_QUICK_EDIT_MODE is not set, then we need
+        // to ask the terminal to send us mouse events - in all other cases, terminal should
+        // stop sending us mouse events
+        if ((oldMouseMode != newMouseMode) ||
+            (oldQuickEditMode != newQuickEditMode))

```

Comment:
I'm not sure this is quite right -- we only ever care about the current final status of quickedit when mouse mode is being toggled, right?

What we want to know is "*will user input generate mouse records in conhost*?"

You can probably simplify the logic here substantially:

```
const auto oldMouseMode{ !oldQuickEditMode && WI_IsFlagSet(old, mouse) };
const auto newMouseMode{ !newQuickEditMode && WI_IsFlagSet(new, mouse) };
```

"The mouse was enabled prior if quickedit was off and mouse was on" "the mouse is newly-enabled if quickedit is off and mouse is on".

Reply:
Ah yes this way is much less verbose, fixed!